sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/InputsAdapter.java

```diff
@@ -296,32 +299,42 @@ private Optional<SclReportItem> updateSourceDataSetsAndControlBlocks(TExtRef ext
             return warningReportItem(extRef, String.format(MESSAGE_SOURCE_LN_NOT_FOUND, optionalSourceLDevice.get().getXPath()));
         }
 
-        //TODO: Don't forget to test this in issue #84 RSR-433
-        List<ResumedDataTemplate> mxAndStSources = sourceDas.stream()
-            .filter(da -> da.getFc() == TFCEnum.MX || da.getFc() == TFCEnum.ST).toList();
+        return filterValidSourceDas(extRef, sourceDas);
+    }
 
-        Stream<ResumedDataTemplate> sourceDaToProcess;
+    private Optional<SclReportItem> filterValidSourceDas(TExtRef extRef, final Set<ResumedDataTemplate> sourceDas) {
+        //TODO: Don't forget to test this in issue #84 RSR-433
+        sourceDas.removeIf(da -> da.getFc() != TFCEnum.MX && da.getFc() != TFCEnum.ST);
         switch (extRef.getServiceType()) {
-            case GOOSE:
-            case SMV:
-                sourceDaToProcess = mxAndStSources.stream().filter(FcdaCandidates.SINGLETON::contains);
-                break;
-            case REPORT:
-                //TODO: In issue RSR-608, replace with mxAndStSources.filter(...)
-                sourceDaToProcess = Stream.empty();
-                break;
-            case POLL:
-            default:
+            case GOOSE, SMV -> sourceDas.removeIf(Predicate.not(FcdaCandidates.SINGLETON::contains));
+            case REPORT -> {
+                Optional<SclReportItem> filteringError = filterReportSourceDa(extRef, sourceDas);
+                if (filteringError.isPresent()) {
+                    return filteringError;
```

Comment:
I think there is no need for that: if you return an optional there is no need to return the optional after checking if it's empty ?
It's a choice between use optional or the value

Reply:
Indeed. OK to return `filterReportSourceDa` result directly.